README.md

```diff
@@ -429,17 +429,37 @@ misbehavingStream.log();
 
 ### Future
 
-#### `Future#listen(o: Consumer<A>): void`
+#### `Future.of(a: A): Future<A>`
 
-Adds a consumer as listener to a future. If the future has already
-occurred the consumer is immediately pushed to.
+Converts any value into a future that happens immediately.
 
 #### `fromPromise<A>(p: Promise<A>): Future<A>`
 
 Converts a promise to a future.
 
+#### `isFuture(f: any): f is Future<any>`
+
+Returns `true` if `f` is a future and `false` otherwise.
+
+#### `Future#listen(o: Consumer<A>): void`
+
+Adds a consumer as listener to a future. If the future has already
+occurred the consumer is immediately pushed to.
+
 ### Stream
 
+#### `empty: Stream<A>`
+
+Empty stream.
+
+#### `Stream.of(a: A): Stream<A>`
+
+Converts any value into a constant stream that happens immediately.
```

Comment:
As you said, this function doesn't exist and there is no good way to implement it.

Reply:
That's un-intuitive to me. I'd love to put in the documentation why this doesn't exist. For example, why can't it have the same semantics at `Future.of` of `(-Infinity, a)`?

Reply:
That's a good question! You're right that we could impleent `Stream.of(a)` to be equivalent to `[(-Infinity, a)]`. The `Stream.of` function could actually work like that.

But, here's the catch: Hareactive never makes it possible to access a streams past occurrences. Said another way, it is impossible to write code that when executed depends on occurrences that happened earlier. There are two reasons for this.

* It makes it possible to forget occurrences as soon as they occur. We just need to notify any listeners and then the data in the occurrence can be garbage collected.
* It's not very useful. In the cases where one would want to know the last occurrence using a behavior is usually the better choice.

In the light of this the `[(-Infinity, a)]` implementation is not useful. It returns a stream of an occurrence that is always in the past and hence an occurrence that is inaccessible. We could actually implement `of` on `Stream` like this:

```
of<A>(a: A): Stream<A> {
  return empty;
}
```

That would be a valid implementation because the occurrence is in the past, and hence inaccessible anyway.

To sum it up, we could implement `Stream.of` but it would not be useful.

Note that `Future` is different. Here we keep the data once the future has occurred.

Reply:
Cool, makes sense! Appreciate this. 

Why do we keep data once `Future` has occurred?

Reply:
Nevermind, I can see why that would make sense for `Future`s